sdk/azidentity/aad_identity_client.go

```diff
@@ -231,12 +227,13 @@ func (c *aadIdentityClient) createClientCertificateAuthRequest(tenantID string,
 	data.Set(qpScope, strings.Join(scopes, " "))
 	dataEncoded := data.Encode()
 	body := azcore.NopCloser(strings.NewReader(dataEncoded))
-	req := azcore.NewRequest(http.MethodPost, u)
-	req.Header.Set(azcore.HeaderContentType, azcore.HeaderURLEncoded)
-	err = req.SetBody(body)
+	req, err := azcore.NewRequest(ctx, http.MethodPost, azcore.JoinPaths(c.options.AuthorityHost, tenantID, tokenEndpoint))
```

Comment:
Since the join of the paths is already done on line 216, I think you can just use 'u' here instead of doing the join again.

Reply:
Fixed